Branch-SDK/Branch-SDK/Networking/BNCNetworkServiceProtocol.h

```diff
@@ -0,0 +1,80 @@
+//
+//  BNCNetworkServiceProtocol.h
+//  Branch-SDK
+//
+//  Created by Edward Smith on 5/30/17.
+//  Copyright © 2017 Branch Metrics. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
```

Comment:
Probably want to add a couple more details like which variables are absolutely required to define in their implementation and which are optional.